receiver/mongodbatlasreceiver/alerts.go

```diff
@@ -71,26 +109,127 @@ func newAlertsReceiver(logger *zap.Logger, cfg AlertConfig, consumer consumer.Lo
 		}
 	}
 
-	recv := &alertsReceiver{
-		addr:        cfg.Endpoint,
-		secret:      cfg.Secret,
-		tlsSettings: cfg.TLS,
-		consumer:    consumer,
-		wg:          &sync.WaitGroup{},
-		logger:      logger,
+	projectExclusionMap := map[string]bool{}
+	projectInclusionMap := map[string]bool{}
+	for _, pc := range cfg.Projects {
+		for _, exclusion := range pc.ExcludeClusters {
+			projectExclusionMap[projectClusterKey(pc, exclusion)] = true
+		}
+		for _, inclusion := range pc.IncludeClusters {
+			projectInclusionMap[projectClusterKey(pc, inclusion)] = true
+		}
 	}
 
-	s := &http.Server{
-		TLSConfig: tlsConfig,
-		Handler:   http.HandlerFunc(recv.handleRequest),
+	recv := &alertsReceiver{
+		addr:              cfg.Endpoint,
+		secret:            cfg.Secret,
+		tlsSettings:       cfg.TLS,
+		consumer:          consumer,
+		mode:              alertMode(cfg.Mode),
+		projects:          cfg.Projects,
+		retrySettings:     baseConfig.RetrySettings,
+		publicKey:         baseConfig.PublicKey,
+		privateKey:        baseConfig.PrivateKey,
+		wg:                &sync.WaitGroup{},
+		pollInterval:      baseConfig.Alerts.PollInterval,
+		doneChan:          make(chan bool, 1),
+		logger:            logger,
+		projectInclusions: projectInclusionMap,
+		projectExclusions: projectExclusionMap,
```

Comment:
I renamed these variables but mostly because they are keyed via project name and cluster name. 

```go
func projectClusterKey(project ProjectConfig, clusterName string) string {
	return fmt.Sprintf("%s|%s", project.Name, clusterName)
}
```

Probably could be confusion on one of our ends but that was the anticipated solution I had in mind.

Reply:
If that's the case I think we can simplify and get rid of the map idea in the first place since we already have a ProjectConfig available in the `applyFilters` function. The idea with the map was to avoid already known iteration through the exclude/include lists.

Reply:
See comment here that may justify a map, but I think any include/exclude maps should be localized to each project: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14180#discussion_r983611478